plan/statistics/statistics.go

```diff
@@ -336,36 +575,38 @@ func TableFromPB(ti *model.TableInfo, tpb *TablePB) (*Table, error) {
 	if len(tpb.Columns) != len(ti.Columns) {
 		return nil, errors.Errorf("column count not match, expected %d, got %d", len(ti.Columns), len(tpb.Columns))
 	}
+	if len(tpb.Indices) != len(ti.Indices) {
```

Comment:
Yes, it also must be considered by cache statistics. Currently, except in the test suit, no one will use it. In the cache statistics PR, it will be handled by it's caller, by simply drop it and use a pseduo table. I will leave a TODO here in the next PR, and maybe we can have better solutions.